mirgecom/boundary.py

```diff
@@ -1909,59 +1901,73 @@ def __init__(self, free_stream_state=None,
             self._ref_pressure = free_stream_state.pressure
             self._spec_mass_fracs = free_stream_state.cv.species_mass_fractions
 
-        if self._ref_velocity.shape[0] != 2:
+        if dim == 3:
+            raise ValueError("This BC only supports 1 or 2-dimensional inputs.")
+
+        if self._ref_velocity.shape[0] != dim:
             vel_shape = self._ref_velocity.shape[0]
-            raise ValueError(f"Expected 2-dimensional inputs, got {vel_shape}.")
+            raise ValueError(f"Expected {dim}-dimensional inputs, got {vel_shape}.")
```

Comment:
There may be something subtle I am missing here, but why not:

```
dim = self._ref_velocity.shape[0]
if dim == 3:
   yadda yadda
```

Reply:
https://github.com/illinois-ceesd/mirgecom/pull/1002/commits/b5961dd4b1ce9ae0032071ea48949d3d06d3d363